runtime/src/internal/shared.d.ts

```diff
@@ -0,0 +1,27 @@
+export const CONTEXT_KEY: unknown;
+
+export type Params = Record<string, string>;
+export type Query = Record<string, string | string[]>;
+export type Preload = (props: { params: Params, query: Query }) => Promise<any>;
+
+export interface ComponentModule {
+  default: ComponentConstructor | SSRComponent
+  preload?: Preload;
+}
+
+export interface DOMComponent {
+	$set: (data: any) => void;
+	$destroy: () => void;
+}
+
+export interface ComponentConstructor {
+  new(options: { target: unknown, props: unknown, hydrate: boolean }): DOMComponent;
```

Comment:
The actual type for target is Node, which requires the typescript dom library. Unfortunately anything that's in this file needs to work on both node and in the browser.

Reply:
done